### ISSUE TITLE
Transparency in PNGs is rendered as black

### ISSUE THREAD

**Issue by Spenhouet** [jorisschellekens/borb#77]
I did load and store an existing PDF page ([test.pdf](https://github.com/jorisschellekens/borb/files/8155157/test.pdf)) like this:

```python
from typing import Optional
from borb.pdf.document.document import Document
from borb.pdf.pdf import PDF

doc: Optional[Document] = None
with (scratch / 'test.pdf').open('rb') as in_file_handle:
    doc = PDF.loads(in_file_handle)

with open(scratch / "bug-png-transparency.pdf", "wb") as pdf_file_handle:
    PDF.dumps(pdf_file_handle, doc)
```

The PNG images in the PDF are no longer rendered with transparent background and instead show a black background:

| before | after |
| ----- | ------ |
| ![image](https://user-images.githubusercontent.com/7819068/156024908-870dd8f8-c5b2-4cfe-89fa-7068f8edba86.png)  |  ![image](https://user-images.githubusercontent.com/7819068/156024973-58be5bc7-cd60-4656-b29c-821e511b6072.png)  | 

Am I missing some option during read or write?

**Comment by Spenhouet** [comment]
Thanks for the fast feedback. Is there a workaround?

Seems to be the only think not working with our PDFs. Would be a shame if that would prevent us from being able to use borb.

**Comment by jorisschellekens** [comment]
I'm surprised the workaround I implemented in `borb` does not seem to be working. It should have converted all transparent pixels to white pixels (since the default page background is usually white).

I'll have a closer look later this week.

Possible workarounds:
- if you are creating the document, don't insert PNG images. Convert them to JPEG first.
- before storing the document, iterate over all its embedded images, and convert then to JPEG. 

Small question: you mentioned "us" in your last comment. Are you using `borb` in a commercial setting?

Kind regards,
Joris Schellekens

**Comment by Spenhouet** [comment]
> I'm surprised the workaround I implemented in borb does not seem to be working. It should have converted all transparent pixels to white pixels (since the default page background is usually white).

This would be sufficient in this case. I attached the PDF I tested it with.

> if you are creating the document, don't insert PNG images. Convert them to JPEG first.

That would work. 

> before storing the document, iterate over all its embedded images, and convert them to JPEG.

Easier said than done. At least not if one does not know how to.

> Small question: you mentioned "us" in your last comment. Are you using borb in a commercial setting?

We are not using it yet. We were just evaluating tools for our use-case.
But good that you pointed out the license.

> This program is offered under a commercial and under the AGPL license.
For commercial licensing, contact us at https://borbpdf.com.  For AGPL licensing, see below.

I was not aware before that this library has a commercial license.
Our potential usage of the library is pretty limited. As I said previously we are mostly using weasyprint and jinja to render PDFs.
We just need to fill in additional text (like templating but after the PDF was created) without using form fields.
So if the commercial license includes any costs, that will be a show stopper for us.
Then we just have to go with the slightly worse solution using the outdated PyPDF4.
I will still contact you to clarify what your commercial license entails.
